org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/part/DependencyEditorMenuPart.java

```diff
@@ -0,0 +1,94 @@
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+
+/**
+ * This editor part displays all buttons the with which the user can modify the
+ * list of dependencies.
+ */
+public class DependencyEditorMenuPart {
+	private final Composite composite;
+	private final TargetDependencyModel model;
+	private Button addButton;
+	private Button removeButton;
+	private Button updateButton;
+	private Button redoButton;
+	private Button undoButton;
+
+	public DependencyEditorMenuPart(Composite parent, TargetDependencyModel model) {
+		this.composite = new Composite(parent, SWT.NONE);
+		this.composite.setLayout(new GridLayout());
+		this.composite.setLayoutData(new BorderData(SWT.RIGHT));
+		this.model = model;
+		createContent();
+	}
+
+	private void createContent() {
+		ISideEffectFactory factory = WidgetSideEffects.createFactory(composite);
+
+		addButton = newButton(Messages.MavenTargetDependencyEditor_7, Messages.MavenTargetDependencyEditor_12);
+		addSelectionChangedListener(addButton, model::add);
+
+		removeButton = newButton(Messages.MavenTargetDependencyEditor_8, Messages.MavenTargetDependencyEditor_13);
+		addSelectionChangedListener(removeButton, model::remove);
+		factory.create(() -> removeButton.setEnabled(model.canRemove()));
+
+		updateButton = newButton(Messages.MavenTargetDependencyEditor_9, Messages.MavenTargetDependencyEditor_14);
+		addSelectionChangedListener(updateButton, model::update);
+		factory.create(() -> updateButton.setEnabled(model.canUpdate()));
+
+		undoButton = newButton(Messages.MavenTargetDependencyEditor_10, null);
+		addSelectionChangedListener(undoButton, model.getHistory()::undo);
+		factory.create(() -> undoButton.setEnabled(model.getHistory().canUndo()));
+
+		redoButton = newButton(Messages.MavenTargetDependencyEditor_11, null);
+		addSelectionChangedListener(redoButton, model.getHistory()::redo);
+		factory.create(() -> redoButton.setEnabled(model.getHistory().canRedo()));
+
+		// The table viewer should have the same height as all buttons combined
+		model.setHeightHint(composite.computeSize(SWT.DEFAULT, SWT.DEFAULT).y);
+	}
+
+	private Button newButton(String text, String tooltip) {
+		Button button = new Button(composite, SWT.PUSH);
+		button.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, false));
+		button.setText(text);
+		button.setToolTipText(tooltip);
+		return button;
+	}
+
+	/**
+	 * Registers a new selection-change listener for the given button. Whenever the
+	 * button is clicked, a model change has to be performed.
+	 * 
+	 * @param button      One of the buttons of the menu part.
+	 * @param onSelection The method that is called when the button is clicked.
+	 */
+	private void addSelectionChangedListener(Button button, Runnable onSelection) {
+		button.addSelectionListener(SelectionListener.widgetSelectedAdapter(event -> {
+			onSelection.run();
+		}));
+	}
```

Comment:
The fields are only used in the `createContent()` method and can therefore be inlined in that method.
Furthermore, setting the selection-listener can also be done in the `newButton() `method.
```suggestion
public class DependencyEditorMenuPart {
	private final Composite composite;
	private final TargetDependencyModel model;

	public DependencyEditorMenuPart(Composite parent, TargetDependencyModel model) {
		this.composite = new Composite(parent, SWT.NONE);
		this.composite.setLayout(new GridLayout());
		this.composite.setLayoutData(new BorderData(SWT.RIGHT));
		this.model = model;
		createContent();
	}

	private void createContent() {
		ISideEffectFactory factory = WidgetSideEffects.createFactory(composite);

		newButton(Messages.MavenTargetDependencyEditor_7, Messages.MavenTargetDependencyEditor_12, model::add);

		Button removeButton = newButton(Messages.MavenTargetDependencyEditor_8, Messages.MavenTargetDependencyEditor_13,
				model::remove);
		factory.create(() -> removeButton.setEnabled(model.canRemove()));

		Button updateButton = newButton(Messages.MavenTargetDependencyEditor_9, Messages.MavenTargetDependencyEditor_14,
				model::update);
		factory.create(() -> updateButton.setEnabled(model.canUpdate()));

		Button undoButton = newButton(Messages.MavenTargetDependencyEditor_10, null, model.getHistory()::undo);
		factory.create(() -> undoButton.setEnabled(model.getHistory().canUndo()));

		Button redoButton = newButton(Messages.MavenTargetDependencyEditor_11, null, model.getHistory()::redo);
		factory.create(() -> redoButton.setEnabled(model.getHistory().canRedo()));

		// The table viewer should have the same height as all buttons combined
		model.setHeightHint(composite.computeSize(SWT.DEFAULT, SWT.DEFAULT).y);
	}

	/**
	 * Creates a new push-button with the given text, tool-tip and selection-change
	 * listener. Whenever the button is clicked, a model change has to be performed.
	 * 
	 * @param text        the text displayed on the button
	 * @param tooltip     the tooltip displayed for the button
	 * @param onSelection the method that is called when the button is clicked.
	 */
	private Button newButton(String text, String tooltip, Runnable onSelection) {
		Button button = new Button(composite, SWT.PUSH);
		button.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, false));
		button.setText(text);
		button.setToolTipText(tooltip);
		button.addSelectionListener(SelectionListener.widgetSelectedAdapter(event -> onSelection.run()));
		return button;
	}
```